src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionRetentionStrategy.java

```diff
@@ -114,4 +156,55 @@ protected long getIdleMaxTime() {
 
 
     private static Logger LOGGER  = Logger.getLogger(MansionRetentionStrategy.class.getName());
+    
+    private static final QueueLock queueLock = newQueueLock();
+    
+    private static QueueLock newQueueLock() {
+        try {
+            return new ReflectionPost592QueueLock();
+        } catch (NoSuchMethodException e) {
+            return new Pre592QueueLock();
+        }
+    }
+
+    // TODO replace all this with Queue.withLock once Jenkins 1.592+
```

Comment:
This plugin is available for use on prem also and our official line for this month at least is 1.565.3+ even after this month it only goes to 1.580.3+

Sent from my iPhone

> On 9 Jul 2015, at 19:39, Ryan Campbell <notifications@github.com> wrote:
> 
> In src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionRetentionStrategy.java:
> 
> > @@ -114,4 +156,55 @@ protected long getIdleMaxTime() {
> >  
> >  
> >      private static Logger LOGGER  = Logger.getLogger(MansionRetentionStrategy.class.getName());
> > +    
> > +    private static final QueueLock queueLock = newQueueLock();
> > +    
> > +    private static QueueLock newQueueLock() {
> > +        try {
> > +            return new ReflectionPost592QueueLock();
> > +        } catch (NoSuchMethodException e) {
> > +            return new Pre592QueueLock();
> > +        }
> > +    }
> > +
> > +    // TODO replace all this with Queue.withLock once Jenkins 1.592+
> Well, I think you could just upgrade the POM to 1.609 and avoid all this. DEV@cloud should be on 1.609.1 by Monday.
> 
> —
> Reply to this email directly or view it on GitHub.
>